ludwig/features/text_feature.py

```diff
@@ -261,6 +264,36 @@ def get_output_dtype(cls):
     def output_shape(self) -> torch.Size:
         return torch.Size([self.decoder_obj.config.max_sequence_length])
 
+    def update_metrics(
+        self,
+        targets: Tensor,
+        predictions: Dict[str, Tensor],
+        decoded_targets: Optional[List[str]] = None,
+        decoded_predictions: Optional[List[str]] = None,
+    ) -> None:
+        """Updates metrics with the given targets and predictions.
+
+        If decoded_targets and decoded_predictions are provided, as through LLM model types, then additional
+        response-based metrics like BLEU and ROUGE are also computed.
+
+        Args:
+            targets: Tensor with target values for this output feature.
+            predictions: Dict of tensors returned by predictions().
+        """
+        for metric_name, metric_fn in self._metric_functions.items():
+            prediction_key = get_metric_tensor_input(metric_name)
+            if prediction_key != RESPONSE:
+                # Non-RESPONSE metrics don't use decoded texts.
+                metric_fn = metric_fn.to(predictions[prediction_key].device)
+                metric_fn.update(predictions[prediction_key].detach(), targets)
+                continue
+
+            if decoded_targets is not None and decoded_predictions is not None:
+                # RESPONSE metrics cannot be computed if decoded texts aren't provided.
+                # We assume that RESPONSE metrics can only be calculated one example at a time.
```

Comment:
OOC, why do we make this assumption?

Reply:
Ah, this appears to be an incorrect assumption. torchmetrics text docs all use single examples. However, batching `(preds, targets)` works great, consistent with other torchmetrics. Removing the extra for loop.